bot/seasons/valentines/be_my_valentine.py

```diff
@@ -0,0 +1,228 @@
+        role = discord.utils.get(ctx.guild.roles, id=Lovefest.role_id)
+        if id not in [role.id for role in ctx.message.author.roles]:
+            USER_LOVEFEST.append(ctx.author)
+            await user.add_roles(role)
+            await ctx.send("The Lovefest role has been added !")
+        else:
+            await ctx.send("You already have the role !")
+
+    @with_role(Roles.moderator)
+    @commands.command(name='refreshlovefest')
+    async def refresh_user_lovefestlist(self, ctx):
+        """
+        Use this command to refresh the USER_VALENTINE list when the bot goes offline and then comes back online
+        """
+        USER_LOVEFEST.clear()
+        for member in ctx.guild.members:
+            for role in member.roles:
+                if role.id == Lovefest.role_id:
+                    USER_LOVEFEST.append(member)
+        embed = discord.Embed(
+            title="USER_LOVEFEST list updated!",
+            description=f'''The USER_LOVEFEST has been refreshed,`bemyvalentine` and `bemyvalentine dm` commands can now
+                            be used and there are {USER_LOVEFEST.__len__()} members having the lovefest role.''',
+            color=Colours.pink
+        )
+        await ctx.send(embed=embed)
+
+    @commands.cooldown(1, 1800, BucketType.user)
+    @commands.group(name='bemyvalentine', invoke_without_command=True)
+    async def send_valentine(self, ctx, user: typing.Optional[discord.Member] = None, *, valentine_type=None):
+        """
+        This command sends valentine to user if specified or a random user having lovefest role.
+
+        syntax: .bemyvalentine [user](optional) [p/poem/c/compliment/or you can type your own valentine message]
+        (optional)
+
+        example: .bemyvalentine (sends valentine as a poem or a compliment to a random user)
+        example: .bemyvalentine @Iceman#6508 p (sends a poem to Iceman)
+        example: .bemyvalentine @Iceman#6508 Hey I love you, wanna hang around ? (sends the custom message to Iceman)
+        """
+        if ctx.guild is None:
+            # This command should only be used in the server
+            msg = "You are supposed to use this command in the server."
+            return await ctx.send(msg)
+
+        channel = self.bot.get_channel(Lovefest.channel_id)
+
+        if user == ctx.author:
+            # Well a user cant valentine himself/herself.
+            await ctx.send('Come on dude, you cant send a valentine to yourself :expressionless:')
+
+        elif user is None:
+            # just making sure that the random does not pick up the same user(ctx.author)
+            USER_LOVEFEST.remove(ctx.author)
+            user = random.choice(USER_LOVEFEST)
+            USER_LOVEFEST.append(ctx.author)
+
+            if valentine_type is None:
+                # grabs a random valentine -can be a poem or a good message
+                valentine, title = self.random_valentine()
+                embed = discord.Embed(
```

Comment:
This embed generation doesn't need to be repeated for every conditional. The `f"{EMOJI_1} {title} {user.display_name}{EMOJI_2}"` format is exactly the same across them, so you can instead generate `title` inside of the conditions and just have a single embed statement after the logic block.